doc/develop-plugin.md

```diff
@@ -0,0 +1,159 @@
+
+Note: Even if the original plugin is not a federated plugin (has no platform interface package), it is possible to create an unendorsed platform implementation of the plugin by implicitly implementing the plugin's platform channels.
+
+You can create a new plugin from scratch, if the functionality you want to implement is not implemented by any other plugin, or is specific to Tizen. The new plugin can be either a single package plugin or a federated plugin, depending on whether you want to target the Tizen platform only or other platforms as well.
+
+## Create a plugin package
+
+If you're to extend an existing plugin for Tizen, it is common to add the `_tizen` suffix to the package name:
+
+```sh
+flutter-tizen create --template plugin foobar_tizen
+```
+
+Otherwise, follow the `lowercase_with_underscores` style convention to name your package:
+
+```sh
+flutter-tizen create --template plugin plugin_name
+```
+
+Once the package is created, you will be prompted to add some information to its pubspec file. Open the main `plugin_name/` directory in VS Code, locate the `pubspec.yaml` file, and replace the `some_platform:` map with `tizen:` as suggested by the tool. This information is needed by the flutter-tizen tool to find and register the plugin when building an app that depends on the plugin.
+
+```yaml
+The `pubspec.yaml` under the project directory must be updated to support Tizen.
+Add below lines to under the `platforms:` key.
+
+tizen:
+  pluginClass: PluginNamePlugin
+  fileName: plugin_name_plugin.h
+```
+
+The created package contains an example app in the `example/` directory. You can run it using the `flutter-tizen run` command:
+
+```sh
+$ cd plugin_name/example
+$ flutter-tizen run
+```
+
+## Implement the plugin
+
+### 1. Define the package API (`.dart`)
+
+The API of the plugin package is defined in Dart code. Locate the file `lib/plugin_name.dart` in VS Code, and then you will see the `platformVersion` method defined in the plugin main class. Invoking this method will invoke the `getPlatformVersion` method through a method channel named `plugin_name`. This file is not required if you're extending an existing plugin for Tizen, so you can safely remove it.
+
+### 2. Add Tizen platform code (`.cc`)
+
+Note: Before getting started, it is recommended to install the [C/C++ extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) and add the `flutter-tizen/bin/cache/artifacts/engine/common` directory to your workspace in VS Code.
+
+The implementation of the plugin can be found in the `tizen/src/plugin_name_plugin.cc` file. In this file, you will see:
+
+- `PluginNamePluginRegisterWithRegistrar()`: This function is called by an app that depends on this plugin on startup to set up the `plugin_name` channel.
+- `HandleMethodCall()`: This method handles the `getPlatformVersion` method and returns the result to the caller.
```

Comment:
It should also be explained that this is an example and should be replaced with an actual implementation.